assets/js/phoenix_live_view/view.js

```diff
@@ -777,7 +777,7 @@ export default class View {
     let cid = isCid(forceCid) ? forceCid : this.targetComponentID(inputEl.form, targetCtx)
     let refGenerator = () => this.putRef([inputEl, inputEl.form], "change")
     let formData = serializeForm(inputEl.form, {_target: eventTarget.name})
-    if(inputEl.files && inputEl.files.length > 0){
+    if(inputEl.files && inputEl.files.length > 0 && inputEl.getAttribute(this.binding(PHX_HOOK)) === 'Phoenix.LiveFileUpload'){
```

Comment:
We have a helper we can use here :)

```suggestion
    if(DOM.isUploadInput(inputEl) && inputEl.files && inputEl.files.length > 0){
```

Reply:
Perfect!